esrally/mechanic/launcher.py

```diff
@@ -232,9 +164,10 @@ def start(self, node_configurations):
             nodes.append(cluster.Node(p, host_name, node_name, t))
         return nodes
 
-    def _start_process(self, cmd, node_name):
-        return _start(subprocess.Popen(shlex.split(cmd),
-                                       stdout=subprocess.PIPE, stderr=subprocess.STDOUT, stdin=subprocess.DEVNULL), node_name)
+    def _start_process(self, cmd):
+        subprocess.Popen(shlex.split(cmd), close_fds=True)
+        #TODO:  wait_for_docker_pidfile?
```

Comment:
@dliappis can we assume that if `docker-compose` returns (with exit code `0`), Elasticsearch is about to startup? (i.e. no need to wait for the PID file)?

Reply:
I am afraid it's a "it depends" answer :)

With docker-compose it's possible to have healthchecks and depending on how they are defined we can safely assume ES is actually listening; this is done for example in https://github.com/elastic/rally/blob/69cb36d65f35833632dbddabbb043f387039f8bf/docker/docker-compose-tests.yml#L26-L30 and the other service, which can be a sidecar, [depends on it](https://github.com/elastic/rally/blob/69cb36d65f35833632dbddabbb043f387039f8bf/docker/docker-compose-tests.yml#L11-L13).

Reply:
@dliappis I think this is entirely in our control as the only way to start the Docker image is via https://github.com/elastic/rally/blob/master/esrally/resources/docker-compose.yml - So we cannot assume it is already listening (which is ok) but I wonder whether we can assume that the Elasticsearch process has been started already?